exec/csnex/Number.cs

```diff
@@ -34,6 +34,38 @@ public static Number Subtract(Number x, Number y)
             return new Number(x.val - y.val);
         }
 
+        public static Number Multiply(Number x, Number y)
+        {
+            return new Number(x.val * y.val);
+        }
+
+        public static Number Divide(Number x, Number y)
+        {
+            return new Number(x.val / y.val);
+        }
+
+        public static Number Pow(Number x, Number y)
+        {
+            if (y.IsInteger() && !y.IsNegative()) {
+                UInt32 iy = number_to_uint32(y);
+                Number r = new Number(1);
+                while (iy != 0) {
+                    if ((iy & 1) == 1) {
+                        r = Multiply(r, x);
+                    }
+                    x = Multiply(x, x);
+                    iy >>= 1;
+                }
+                return r;
+            }
+            return new Number(Decimal.ToInt64(x.val) ^ Decimal.ToInt64(y.val));
```

Comment:
Just FYI at some point you'll need to handle noninteger exponents. But this is fine for now if it makes some tests pass.